components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/RedisSessionDataStore.java

```diff
@@ -0,0 +1,277 @@
+        // Empty method
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    public void removeExpiredSessionData() {
+        // Empty method
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    public void removeSessionData(String key, String type, long nanoTime) {
+
+        Jedis jedis;
+        if (!enablePersist) {
+            return;
+        }
+        String redisKey = key + Constants.DIVIDER + type;
+        boolean tempStore = getSessionStoreType(type);
+        if (tempStore) {
+            redisKey = redisKey + Constants.DIVIDER + Constants.TEMPSTORE;
+        }
+
+        try {
+            jedis = getJedisInstance().getResource();
+            if (jedis.exists(redisKey)) {
+                jedis.del(redisKey);
+            }
+            String objectKey = redisKey + Constants.DIVIDER + Constants.OBJECT;
+            if (jedis.exists(objectKey)) {
+                jedis.del(objectKey);
+            }
+            jedis.close();
+
+        } catch (JedisException e) {
+            log.error("Error while storing DELETE operation session data", e);
+        }
+    }
+
+    @Override
+    /**
+     * @deprecated This is now run as a part of the {@link #removeExpiredSessionData()} due to a possible deadlock as
+     * mentioned in IDENTITY-5131
+     */
+    public void removeExpiredOperationData() {
+        // Empty method
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    public SessionContextDO getSessionContextData(String key, String type) {
+
+        Jedis jedis;
+        String redisKey = key + Constants.DIVIDER + type;
+        if (!enablePersist) {
+            return null;
+        }
+        boolean tempStore = getSessionStoreType(type);
+        if (tempStore) {
+            redisKey = redisKey + Constants.DIVIDER + Constants.TEMPSTORE;
+        }
+        try {
+            jedis = getJedisInstance().getResource();
+            if (jedis.exists(redisKey)) {
+                long nanoTime = Long.parseLong(jedis.hget(redisKey, Constants.NANO_TIME));
+                String objectKey = redisKey + Constants.DIVIDER + Constants.OBJECT;
+                Object blobObject = deserialize(jedis.get(objectKey.getBytes()));
+                jedis.close();
+                return new SessionContextDO(key, type, blobObject, nanoTime);
+            }
+            return null;
+        } catch (JedisException e) {
+            log.error("Error while retrieving session data", e);
+        }
+        return null;
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    public void persistSessionData(String key, String type, Object entry, long nanoTime, int tenantId) {
+
+        long validityPeriodNano = 0L;
+        String objectKey;
+        Long expireTime;
+        String redisKey;
+        boolean tempStore = getSessionStoreType(type);
+        redisKey = key + Constants.DIVIDER + type;
+        if (!enablePersist) {
+            return;
+        }
```

Comment:
This logic should be separate from the method logic. The `persistSessionData` method should be called upon performing this check.